default/scripting/ship_parts/FighterHangar/FT_HANGAR_1.focs.txt

```diff
@@ -7,7 +7,15 @@ Part
     damage = 1
     combatTargets = And [
         [[COMBAT_TARGETS_VISIBLE_ENEMY]]
-        Fighter
+        TopmostMatches [
+            And [
```

Comment:
This is intended to match bombers first no other fighters. Not how to determine that the target is a bomber. Maybe hardcoded attack value is ok (but will be wrong with tech advances). I could try to somehow look up the design of the ship the fighter started from and see if it contains the right hangar parts. Or somehow tag the bomber fighters when launching in cpp. Or it could try to figure out what the highest fighter Attack value is and target those first

Reply:
"Bombers" are fighters with damage > 1

Reply:
damage > 1 includes Fighters

Reply:
and also includes "Interceptors" after the first tech is researched (and probably(?) good weapon species interceptors are also damage > 1)

Reply:
OK, I thought they just got more shots. Apparently not, though. But in that case, if interceptors only target fighters, then the extra damage is pretty useless.

Reply:
Yes, more shots/hangar capacity makes more sense for interceptors. Maybe i could take N most dangerous enemy fighters, where N is the count of own interceptors * 1.2

Reply:
Implemented a spread macro for targeting  Still figuring out the numbers, but 1.2 seems ok as heuristic. Could be adapted to species traits WIDE_TARGET_SPREAD for laid back or similar NARROW_TARGET_SPREAD for neurotic species.

Reply:
I think this is better as written now, with the simpler condition. The 1.2 factor and taking N most is overly complicated when written with the subcondition.

If you wanted, though, you could extended `OrderedAlternativesOf` to take an optional `number` or `atleast` parameter, which indicates the number of matches that need to have been found to stop trying more alternate conditions. Not sure if that should be adding more objects to the matches from each successively tested subcondition, or only matching with a subcondition if at least the minimum objects match the subcondition, though.